python/vimsence.py

```diff
@@ -261,3 +265,49 @@ def get_directory():
     '''
 
     return re.split(r'[\\/]', vim.eval('getcwd()'))[-1]
+
+def get_dir_path():
+    '''Get absolute path of current dir in form of list
+    :return: list of directory in order
+    eg: INPUT : "/home/anurag/.vim/bundle/vimsence/python"
+        OUTOUT: ['home', 'anurag', '.vim', 'bundle', 'vimsence', 'python']
+    '''
+    return re.split(r'[\\/]', vim.eval('getcwd()'))[1:]
+    
+def get_git_info():
+    ''' This funtion return a Git Repo link and the Repo dir name in form of list
+    else return None if it is not able to extract the Repo Url
+    '''
+
+    dir_list = get_dir_path()
+    url = None
+    dir_name = None
+
+    for i in range(len(dir_list), 0, -1):
+        if dir_list[i-1] == os.environ['USER'] or dir_list[i-1] == 'root':
+            break
+
+        full_path = "/" + "/".join(dir_list[0:i])
+        if '.git' in os.listdir(full_path):
+            with open(full_path + "/.git/config", 'r') as f:
+                for line in f:
+                    dir_name = dir_list[i-1]
+                    # Extract the url from HTTPS clone
+                    if re.search("url = https:", line):
+                        url = re.findall('https://.*', line)[0].removesuffix(".git").replace('anurag3301@', '')
```

Comment:
Your username is still hardcoded in this replacement

Reply:
Oh sorry, I'm a newbie😅, Let me fix it.

Reply:
Fixed this issue. Added much more robust regex in commit  2804144